juriscraper/opinions/united_states/state/miss.py

```diff
@@ -1,100 +1,77 @@
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.domain = "https://courts.ms.gov"
         self.court_id = self.__module__
+        self.url = (
+            "https://courts.ms.gov/appellatecourts/docket/gethddates.php"
+        )
         self.method = "POST"
-        self.number_of_dates_to_process = 5
-        self.pages = {}
-        self.parameters = {"crt": self.get_court_parameter()}
-        self.status = "Published"
-        self.url = f"{self.domain}/appellatecourts/docket/gethddates.php"
+        self.date = None
+        self.court = "SCT"
 
-    def get_court_parameter(self):
-        return "SCT"
+    def _download(self, request_dict={}) -> None:
 
-    """Retrieve dates for which there are case listings.
-    This site's architecture is no bueno. We have to issue
-    a POST request to this page to get a array (in the form
-    of a string) or dates that have cases associated with
-    them.
-    """
-
-    def _download(self, request_dict={}):
-        dates_page = super()._download(request_dict)
-        self.parse_date_pages(dates_page)
-
-    """Keep track of the most recent N date pages.
-    We dont want to crawl all the way back to 1996, so we only
-    parse the most recent [self.number_of_dates_to_process]
-    number of date pages.  Since cases are usually published
-    once a week, this means scraping about the most recent
-    months worth of cases.
-    """
-
-    def parse_date_pages(self, dates_page):
-        # For testing, each example file should be a specific sub-date page,
-        # like https://courts.ms.gov/Images/HDList/SCT02-27-2020.html
         if self.test_mode_enabled():
-            # date below is arbitrary and doesnt matter, it just
-            # needs to be static for testing to work
-            self.pages["2020-02-28"] = dates_page
+            self.date = "02/28/2020"  # some random date for testing
+            self.html = super()._download()
+            self._process_html()
             return
-        for date in self.get_dates_from_date_page(dates_page):
-            url = "{}/Images/HDList/SCT{}.html".format(
-                self.domain,
-                datetime.date.strftime(date, "%m-%d-%Y"),
-            )
-            page = self._get_html_tree_by_url(url)
-            self.pages[f"{date}"] = page
-
-    """Convert string of dates on page into list of date objects.
-    """
 
-    def get_dates_from_date_page(self, dates_page):
-        dates = []
-        substrings = dates_page.text_content().split('"')
-        for substring in substrings:
-            try:
-                dates.append(convert_date_string(substring))
-            except ValueError:
-                pass
-        dates.sort(reverse=True)
-        return dates[: self.number_of_dates_to_process]
+        self.parameters = {"court": self.court}
+        dates = super()._download(request_dict)
+        dates = literal_eval(dates.xpath("//p")[0].text_content())
```

Comment:
I've never used `literal_eval`, but I read the docs. What's it accomplishing here?

Reply:
The api returns a list of dates as. A string.  It just converts the str list to list

Reply:
It’s quite a nice tool

Reply:
is it json? If so, I'd sort of expect `json` to be the library to use there.

Reply:
I think it reruns a byte string of the array of dates for the calendar if I’m remembering

Reply:
In json

Reply:
Yeah, so I'd use json rather than literal_eval, which is more esoteric. Usually the ast library — at least as I've seen it — is used to parse *actual* python code. Like, in a tool like `black` that has to make sense of your code, `ast` is the perfect tool. If we're dealing with json on the wire, I think the ast library isn't quite the right tool

Reply:
Sounds good.

Reply:
Can I just share with you what is actually returned ...

b'<p>["12-31-2020","12-31-2015","12-31-2009","12-31-2003","12-31-1998","12-31-1997","12-30-2010","12-30-2004","12-30-1999","12-29-2016","12-29-2011","12-29-2005","12-29-2000","12-28-2017","12-28-2006","12-27-2018","12-27-2012","12-27-2007","12-27-2001","12-26-2019","12-26-2013","12-25-2014","12-25-2008","12-25-2003","12-24-2020","12-24-2015","12-24-2009","12-23-2010","12-23-2004","12-23-1999","12-22-2016","12-22-2011","12-22-20.... ]</p>

Reply:
I see now - parsed and loaded in via json.  You taught me this a while back but it came back now.